Source/Scene/BillboardCollection.js

```diff
@@ -1015,6 +1053,41 @@ define([
      * @exception {RuntimeError} image with id must be in the atlas.
      */
     BillboardCollection.prototype.update = function(context, frameState, commandList) {
+        var i;
+        var j;
+
+        var startTime = getTimestamp();
+        var timeSlice = this._clampTimeSlice;
+        var endTime = startTime + timeSlice;
+
+        var tileList = this._renderedTileList;
+        while (tileList.length > 0) {
+            var tile = tileList[0];
+            var customData = tile.customData;
+            var customDataLength = customData.length;
+
+            var timeSliceMax = false;
+            for (i = this._lastTileIndex; i < customDataLength; ++i) {
+                var data = customData[i];
+                var object = data.object;
+                if (defined(object) && object instanceof Billboard) {
```

Comment:
Why not let billboard add its own list to the custom data to avoid this basically linear search using `instanceof`?  For example, see uses of [Context._cache](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Renderer/Context.js#L337).